### PR TITLE
Modifications for NuttX 11

### PR DESCRIPTION
**Pull request by robertobucher**
Modifications to compile microros under NuttX 11.

### CODE REVIEW

**Comment by robertobucher** [comment]
I can^t test the provided example yet! Compilation under Humble and NuttX 11.0 is now ok.

File: README.md [hunk 1]
```diff
@@ -7,6 +7,13 @@
...
 | [![Galactic](https://github.com/micro-ROS/micro_ros_nuttx_app/actions/workflows/ci.yml/badge.svg?branch=galactic&event=schedule)](https://github.com/micro-ROS/micro_ros_nuttx_app/actions/workflows/ci.yml) | [![Rolling](https://github.com/micro-ROS/micro_ros_nuttx_app/actions/workflows/ci.yml/badge.svg?branch=main&event=schedule)](https://github.com/micro-ROS/micro_ros_nuttx_app/actions/workflows/ci.yml) | [![Humble](https://github.com/micro-ROS/micro_ros_nuttx_app/actions/workflows/ci.yml/badge.svg?branch=humble&event=schedule)](https://github.com/micro-ROS/micro_ros_nuttx_app/actions/workflows/ci.yml) |
 
+The last modifications allow to compile the library and the app under NUttX 11. The provided example now compiles, but it should be tested again.
+
+Modifications are only for Humble.
+
+I provided some additional tests, successfully performed on a NUCLEO-F746ZG board, on a new github repository (https://github.com/robertobucher/microros_nuttx).
+
+---------------------------------------------------
```
Comment (pablogs9) [inline README.md:16]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -7,6 +7,13 @@
...
 
 This component has been tested in Nuttx 10.1 and Nuttx 10.3.
```
Comment (pablogs9) [inline README.md:18]:
```suggestion
This component has been tested in Nuttx 10.1, Nuttx 10.3 and Nuttx 11.
```

File: micro_ros_lib/patch1 [hunk 3]
```diff
@@ -0,0 +1,23 @@
+--- micro_ros_src/src/rcutils/src/filesystem.c	2023-01-01 10:46:25.425650951 +0100
++++ /home/bucher/sviluppo/GITHUB/ROS2_e_microROS/rcutils/src/filesystem.c	2022-12-28 14:34:59.020737985 +0100
```
Comment (pablogs9) [inline micro_ros_lib/patch1:2]: Local path

**Pull request closed without merging**